internal/plugin/index.go

```diff
@@ -0,0 +1,87 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/kubeshop/botkube/pkg/multierror"
+)
+
+// Type represents the plugin type.
+type Type string
+
+const (
+	// TypeSource represents the source plugin.
+	TypeSource Type = "source"
+	// TypeExecutor represents the executor plugin.
+	TypeExecutor Type = "executor"
+)
+
+type (
+	// Index defines the plugin repository index.
```

Comment:
Remember we realized the cache is not refreshed even we added new plugins. We would implement 2 options;
- Use etag and if-modified-since if github provides to refresh our cache as a first step once we fetched the index during startup
- Since you already a Index data structure, you can implement a function Exists that checks plugin, if not exist, you can refresh cache
- Or you can simply retry index fetch if you get non-ok from plugin fetch response

Reply:
the simpler the better for me, thanks :)